### PR TITLE
Refactoring Files

### PR DESCRIPTION
**Pull request by CharukaK**
## Purpose
- All the sample files were in .js format renamed all of them to .jsx extension
- Added doc comments to the sample files
- Updated the description in the README.md file

## Test environment
Node JS v8.5.0, npm v5.3.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,72 @@
...
+```
+
+In order to build and run the samples navigate to the samples directory in the main directory and run,
+```bash
+    npm install
+    npm run samples
+```
+
+## Data table
+React VizGrammar required you to arrange your source dataset in a tabular way similar to follwing format.
```
Comment (TanyaM) [inline README.md]: Requires

File: README.md [hunk 2]
```diff
@@ -1 +1,72 @@
...
+            "names" : ["rpm","torque","horsepower", "EngineType"],
+            "types" : ["linear","linear", "ordinal","ordinal"]
+    };
+    data = [
+          [8000, 75, 120, "Piston"], [9000, 81, 130, "Rotary"]]
+    ];
+```
+
+
+metadata.names is an array consists of column names/fields of the table where metadata.types records their types (ordinal (C) or linear).
```
Comment (TanyaM) [inline README.md]: What does ordinal (C) means ?

File: README.md [hunk 3]
```diff
@@ -1 +1,72 @@
...
+names and types are aligned together in a way that "Coulmn1" => 'ordinal' and "Coulmn2" => 'linear' and so on.
+
+data section is a collection of arrays of data rows. Single row is stored as an array and their element order follows the order of metadata.names.
+
+##Chart Config
```
Comment (TanyaM) [inline README.md]: Formatting issue. Need a space after two hashes.

File: README.md [hunk 4]
```diff
@@ -1 +1,72 @@
...
+
+Once the data structure is ready, React VizGrammar will try to render a chart out of it.
+
+Users has to give additional parameters to the gadget using config object. It is a JSON object that has well known configuration properties. For example, User can plot a bar chart by simply giving the chart type in the object as 'bar' in the configuration
```
Comment (TanyaM) [inline README.md]: Grammer mistake. (should be corrected as Users have to)

File: README.md [hunk 5]
```diff
@@ -1 +1,72 @@
...
+E.g Following is the bare minimum set of configurations require to draw a bar chart from above tabular data format.
+```javascript
+let config = {
+    x : "rpm",
+    charts : [{type: "line",  y : "torque", color: "EngineType"}],
+    maxLength: 10,
+    width: 400,
+    height: 200
+}
+```
```
Comment (TanyaM) [inline README.md]: In overall, fix following in README.md
1. Grammar issues
2. Punctuation issues
3. Capitalization of sentence starting
4. Get overall content improved (You can get help of a doc team member if required.)

File: samples/charts/AreaChartConfig.jsx [hunk 6]
```diff
@@ -82,6 +67,23 @@ export default class AreaChartConfigSample extends React.Component {
...
 
+    componentDidMount() {
+        this.interval_id = setInterval(() => {
+            this.setState({
+                data: [
+                    [this.state.timer, this.state.timer === 20 ? null : Math.random() * 100, 10, 'piston'],
+                    [this.state.timer, Math.random() * 100, 10, 'rotary']
+                ],
+                data2: [
+
```
Comment (TanyaM) [inline samples/charts/AreaChartConfig.jsx]: unnecessary new line.

File: src/components/BasicChartHelper.jsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { VictoryLine, VictoryPortal, VictoryScatter, VictoryTooltip, VictoryArea, VictoryBar } from 'victory';
+
+export function generateLineOrAreaChartComponent(config, chartIndex, onClick) {
```
Comment (TanyaM) [inline src/components/BasicChartHelper.jsx]: Why does a helper class has the chart generating logic? Is the class name correct?

File: src/components/BasicChartHelper.jsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                        pointerLength={4}
+                        cornerRadius={2}
+                        flyoutStyle={{ fill: '#000', fillOpacity: '0.8', strokeWidth: 0 }}
+                        style={{ fill: '#b0b0b0' }}
+                    />
+                }
+                size={(
+                    config.charts[chartIndex].style ?
+                        config.charts[chartIndex].style.markRadius || 4 :
+                        4
```
Comment (TanyaM) [inline src/components/BasicChartHelper.jsx]: What is 4? Move magic numbers to constants.

File: README.md [hunk 9]
```diff
@@ -1 +1,81 @@
-# react-VizGrammar
+# React-VizGrammar
+
+React VizGrammar is a wrapper around Victory JS and it makes charting easier by adding boilerplate code so that 
+designers and developers can get started and set it up in a few minutes.
+
+A chart can be embedded in a React environment simply by using the VizG react component.
+```jsx
+    import VizG extend
```
Comment (TanyaM) [inline README.md]: Correct the import

File: README.md [hunk 10]
```diff
@@ -1 +1,81 @@
...
+## Build Process
+
+#### Prerequisites
+- NPM
+- Node 
+
+These prerequisites must be installed before proceeding with the build process. 
+
+#### Build the Library
+In order build React-VizGrammar from the sources, Get a clone from this repository then run the following commands in the terminal from the react-vizgrammar directory.
```
Comment (TanyaM) [inline README.md]: ...repository and then...

File: src/components/BasicChart.jsx [hunk 11]
```diff
@@ -1,82 +1,61 @@
...
+import { getBarComponent, getBrushComponent, getLegendComponent, getLineOrAreaComponent } from './ComponentGenerator.jsx';
 
-const LEGEND_DISABLED_COLOR = 'grey';
-
-/**
- * React component required to render Bar, Line and Area Charts.
- */
-export default class BasicCharts extends React.Component {
-
+export default class BasicChartRef extends React.Component {
```
Comment (TanyaM) [inline src/components/BasicChart.jsx]: As a best practice file name should be equal to the class name.

File: src/components/BasicChart.jsx [hunk 12]
```diff
@@ -87,174 +66,263 @@ export default class BasicCharts extends React.Component {
...
-                const min = Math.min.apply(null, dataSets[dataSetName].map(d => d.x));
+        this.setState(plotData);
 
-                if (xScale === 'linear') {
-                    if (xScale === 'linear' && xDomain[0] !== null) {
-                        if (min > xDomain[0]) {
-                            xDomain[0] = min;
-                            this.xRange[0] = min;
-                        }
+        console.info(this.state);
```
Comment (TanyaM) [inline src/components/BasicChart.jsx]: This line will not be executed since its after a setState

File: samples/charts/ChartWrapper.jsx [hunk 13]
```diff
@@ -21,7 +21,10 @@ import PropTypes from 'prop-types';
 import AppBar from 'material-ui/AppBar';
 import { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText } from 'material-ui/Card';
 import FlatButton from 'material-ui/FlatButton';
-
+import { Link } from 'react-router-dom';
+/**
```
Comment (this) [inline samples/charts/ChartWrapper.jsx]: put a new line between `import`s and the class level jsdoc

File: samples/charts/ScrollToTop.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+import React, { Component } from 'react';
+import {withRouter} from 'react-router-dom';
+
+class ScrollToTop extends Component {
```
Comment (this) [inline samples/charts/ScrollToTop.js]: Please explain the purpose of this React component.
To me this look like a behavior of a component(s), father a component it self.

File: src/components/ComponentGenerator.jsx [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+                <button
+                    onClick={() => {
+                        reset(xRange);
+                    }}
+                >
+                    Reset
+                </button>
+            </div>
+            <div
+                style={{ width: '90%', display: 'inline', float: 'right' }}
```
Comment (this) [inline src/components/ComponentGenerator.jsx]: code formatting??
```jsx
<div style={{ width: '90%', display: 'inline', float: 'right' }} >
```

File: README.md [hunk 16]
```diff
@@ -1 +1,81 @@
...
+designers and developers can get started and set it up in a few minutes.
+
+A chart can be embedded in a React environment simply by using the VizG react component.
+```jsx
+    import VizG from 'react-vizgrammar';
+
+    <VizG config={config} data={data} metadata={metadata} />
+``` 
+where the props
+- ```config``` is the widget configurations specified as a JSON object.
```
Comment (this) [inline README.md]: instead of \```config``` use \`config` for inline code snippets

File: samples/charts/ScrollToTop.js (hunk 14 above)
Comment (CharukaK) [reply samples/charts/ScrollToTop.js]: https://github.com/ReactTraining/react-router/blob/v4.2.1/packages/react-router-dom/docs/guides/scroll-restoration.md

Comment (this) [reply samples/charts/ScrollToTop.js]: ok

**Pull request merged**